### PR TITLE
test(nodejs): add test_all_types.test.ts

### PR DESCRIPTION
**Pull request by Mause**
This is mostly a port from the python test_all_types, with the addition of switching to using the BigInt class for the larger integer types. Worth discussing whether we make this breaking change, or put it behind a setting

### CODE REVIEW

File: tools/nodejs/src/statement.cpp [hunk 1]
```diff
@@ -166,7 +166,11 @@ static Napi::Value convert_col_val(Napi::Env &env, duckdb::Value dval, duckdb::L
 		value = Napi::Number::New(env, duckdb::DoubleValue::Get(dval));
 	} break;
 	case duckdb::LogicalTypeId::HUGEINT: {
-		value = Napi::Number::New(env, dval.GetValue<double>());
+		auto val = duckdb::HugeIntValue::Get(dval);
+		auto negative = val.upper < 0;
+		auto upper = negative ? val.upper * -1 : val.upper; // remove signing bit
```
Comment (Mause) [inline tools/nodejs/src/statement.cpp:171]: There's something subtly wrong with this long, causing the negative result to be out by two, but I'm not sure what it is

File: tools/nodejs/test/test_all_types.test.ts [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    null,
+  ],
+};
+
+const suite = describe("test_all_types", () => {
+  before(async function () {
+    const all_types = await get_all_types();
+
+    for (const cur_type of all_types) {
+      // FIXME: these currently have too high a precision to be tested
```
Comment (Mause) [inline tools/nodejs/test/test_all_types.test.ts:202]: Node doesn't have a builtin Decimal class, so while these are returned, the results are way off. Not sure if we should write a simple wrapper around the BigInt class and use that, return them as strings, or just live with the inaccuracy

File: tools/nodejs/src/statement.cpp (hunk 1 above)
Comment (Tishj) [reply tools/nodejs/src/statement.cpp:171]: Maybe we're missing this step that's done in ToString if the hugeint is negative:
```c++
	static void NegateInPlace(hugeint_t &input) {
		if (input.upper == NumericLimits<int64_t>::Minimum() && input.lower == 0) {
			throw OutOfRangeException("HUGEINT is out of range");
		}
		input.lower = NumericLimits<uint64_t>::Maximum() - input.lower + 1;
		input.upper = -1 - input.upper + (input.lower == 0);
	}
```

Comment (Mause) [reply tools/nodejs/src/statement.cpp]: That was it, cheers!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**